### PR TITLE
Add WorkOS integration

### PR DESCRIPTION
**Pull request by janbarasek**

### CODE REVIEW

File: libs/api/utils/src/lib/api-utils.ts [hunk 1]
```diff
@@ -1,21 +1,53 @@
...
-import type { ApiKeyReportUsageRequest, ApiUsageReport, ApiUsageRequest } from '../typescript'
+import { ERROR_LIST } from '../errorList'
+import { errorResponseModel } from '../model'
+import type { ApiKeyReportUsageRequest, ApiUsageReport, ApiUsageRequest, ExtendedRequest } from '../typescript'
 
 let API_KEY_TO_ID_CACHE: { [key: string]: bigint } = {}
 
-let API_KEY_CONTEXT: string | undefined
+export const apiCommonGuard = new Elysia().guard({
+  as: 'global',
```
Comment (BleedingDev) [inline libs/api/utils/src/lib/api-utils.ts:13]: **thought:** If we change from `global` to `scoped` we might not need `if (path.startsWith('/swagger')) return`.

File: libs/api/utils/src/lib/api-utils.ts [hunk 2]
```diff
@@ -1,21 +1,53 @@
...
+  response: {
+    401: errorResponseModel,
+    404: errorResponseModel,
+    429: errorResponseModel,
+    500: errorResponseModel,
+  },
+  detail: {
+    tags: ['v1'],
```
Comment (BleedingDev) [inline libs/api/utils/src/lib/api-utils.ts:21]: **suggestion:** As this is guard, it shouldn't have tag of "v1".

File: libs/auth/src/user.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { prisma } from '@nmit-coursition/db'
+import type { cas__user } from '@prisma/client'
```
Comment (BleedingDev) [inline libs/auth/src/user.ts:2]: Interesting, it's the first time I see such import from Prisma.

File: libs/auth/src/user.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        organisation_id: organisationId,
+        email: userData.email,
+        inserted_date: new Date(userData.createdAt),
+        updated_date: new Date(userData.updatedAt),
+        synced_date: new Date(),
+        workos_id: userData.id,
+      },
+    }))
+
+  return await prisma.cas__user.update({
```
Comment (BleedingDev) [inline libs/auth/src/user.ts:32]: **suggestion:** We shouldn't await in return, it's already a promise, we can just return it without awaiting.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: libs/auth/src/user.ts (hunk 3 above)
Comment (janbarasek) [reply libs/auth/src/user.ts:2]: It is mine strict-defined response

File: libs/api/utils/src/lib/api-utils.ts (hunk 2 above)
Comment (janbarasek) [reply libs/api/utils/src/lib/api-utils.ts:21]: It has been copy-pasted from original endpoint implementation.

File: libs/auth/src/typescript.ts
Comment (BleedingDev) [inline libs/auth/src/typescript.ts:1]: **suggestion:** I don't think constant should be in `typescript.ts` file, that should be purely for typescript things (like types, interfaces etc.)

**Review by BleedingDev — APPROVED** [review]

**Pull request merged**